### ISSUE TITLE
Better  tf.contrib.estimator.early_stopping (Feature request)

### ISSUE THREAD

**Issue by ElchananHaas** [tensorflow/tensorflow#21555]
### System information
N/A

### Describe the problem
The current early stopping code doesn't take into account any trends in the loss, and therefore can be thrown off by a single low loss. It would be better to use an exponential moving average of the loss to compute the lowest loss, as that will be less likely to be thrown off by a single very low loss.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by Elch123** [comment]
Have I written custom code: Yes
OS Platform and Distribution: Linux 
TensorFlow installed from: pip, 
TensorFlow version: 1.8.0
Bazel version: 0.11.1
CUDA/cuDNN version: 9.0
GPU model and memory: GTX 950 M
Exact command to reproduce:N/A
Mobile deviceN/A

In tensorflows stop if no increase hook, https://www.tensorflow.org/api_docs/python/tf/contrib/estimator/stop_if_no_increase_hook, I find it gives inconsistent results. If there is a single very low loss, then the training will stop too early. What I found worked better was to maintain a running average of the loss, with and stop if that didn't increase beyond a certain number of steps, because that also takes into account trends in the loss.

**Comment by bignamehyp** [comment]
We currently don't have anybody working on this. It would be great if you could help us by working on this and submitting a PR. Let us know if you need further clarification. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by Elch123** [comment]
No, I don't have time to work on this now.

**Comment by tensorflowbutler** [comment]
As this issue has invited community support, please remove the assignee. Otherwise, remove the `community support` label. Thank you.

**Comment by tensorflowbutler** [comment]
As this issue has invited community support, please remove the assignee. Otherwise, remove the `community support` label. Thank you.

**Comment by eyalzk** [comment]
Hi, I would be happy to try implementing this.
Is this still relevant? @facaiy @bignamehyp

**Comment by Elch123** [comment]
It is still relevent. I would love if you would try implementing it.

**Comment by eyalzk** [comment]
I would be happy to. However, I want to wait for a response from one of the TF members so I will know I'm not working on something irrelevant in the project's perspective...

**Comment by rthadur** [comment]
Contrib has been depreciated here in Tensorflow repo and moved to tensorflow/addons , please reopen the issue in https://github.com/tensorflow/addons/issues if this exists in latest version. Thank you
